cohortextractor/main.py

```diff
@@ -1,18 +1,33 @@
 import csv
 import importlib.util
 import inspect
+import shutil
 import sys
 from contextlib import contextmanager
+from pathlib import Path
 
-from .backends import BACKENDS
-from .query_utils import get_column_definitions
+from cohortextractor.backends import BACKENDS
+from cohortextractor.query_utils import get_column_definitions
+from cohortextractor.validate_dummy_data import (
+    DummyDataValidationError,
+    validate_dummy_data,
+)
 
 
-def main(definition_path, output_file, backend_id, db_url):
-    backend = BACKENDS[backend_id](db_url)
+def main(definition_path, output_file, backend_id, db_url, dummy_data_file=None):
     cohort = load_cohort(definition_path)
-    results = extract(cohort, backend)
-    write_output(results, output_file)
+
+    if dummy_data_file:
+        results = None
+        if Path(output_file).suffixes != dummy_data_file.suffixes:
```

Comment:
Could we include this validation in`validate_dummy_data`? That way we can keep everything in `main()` at a similar, high level of abstraction and all the details hidden in the individual methods.